calendar/src/main/java/com/minuk/calendar/PetCalendarView.kt

```diff
@@ -0,0 +1,304 @@
+package com.minuk.calendar
+
+import android.content.Context
+import android.graphics.Color
+import android.util.AttributeSet
+import android.util.Log
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import android.widget.LinearLayout
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.core.content.res.ResourcesCompat
+
+import androidx.recyclerview.widget.RecyclerView
+import com.minuk.calendar.databinding.ItemCalendarBinding
+
+import com.minuk.calendar.databinding.ViewCalendarBinding
+
+import java.util.*
+import kotlin.math.max
+import kotlin.math.min
+
+class PetCalendarView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = 0
+) : LinearLayout(context, attrs, defStyleAttr) {
+
+    private lateinit var calendarBinding: ViewCalendarBinding
+
+    //calendar data property
+    private val calendar = Calendar.getInstance()
+    private var toDay = 0
+    private var startDayOfWeek = 0
+    private var endDayOfCurrentMonth = 31
+    private var endDayOfLastMonth = 31
+    private var daysCount = DEFAULT_DAYS_COUNT
+
+    //custom view property
+    private var viewWidth = 0
+    private var viewHeight = 0
+    private var headerViewHeight = 0f
+    private var itemViewRatio = DEFAULT_ITEM_DIMENS_RATIO
+
+    //Ui data property
+    private var weekendColor = DEFAULT_WEEKEND_COLOR
+    private var weekdaysColor = DEFAULT_WEEKDAYS_COLOR
+    private var todayColor = DEFAULT_TODAY_COLOR
+
+    private var eventHandler: PetCalendarEventHandler? = null
+
+    private var calendarAdapter: PetCalendarAdapter? = null
+
+    init {
+        initLayout(context)
+        initHeaderSize()
+
+        initCalendarData()
+        initDayRecyclerView()
+    }
+
+    private fun initLayout(context: Context) {
+        val inflater = LayoutInflater.from(context)
+
+        calendarBinding = ViewCalendarBinding.inflate(inflater)
+
+        this.addView(calendarBinding.root)
+    }
+
+    private fun initHeaderSize() {
+        headerViewHeight = resources.getDimension(R.dimen.height_header)
+    }
+
+    private fun initCalendarData() {
+        //Log.d(TAG, "year: ${calendar[Calendar.YEAR]} month: ${calendar[Calendar.MONTH]}")
+
+        toDay = calendar[Calendar.DAY_OF_MONTH]
+        calendar.set(Calendar.DAY_OF_MONTH, 1)
+
+        startDayOfWeek = calendar[Calendar.DAY_OF_WEEK] - 1
+        endDayOfCurrentMonth = calendar.getActualMaximum(Calendar.DAY_OF_MONTH)
+
+        calendar.add(Calendar.DAY_OF_MONTH, -1)
+        endDayOfLastMonth = calendar.getActualMaximum(Calendar.DAY_OF_MONTH)
+
+        daysCount = startDayOfWeek + endDayOfCurrentMonth
+        if (daysCount % 7 != 0) {
+            daysCount += 7 - daysCount % 7
+        }
+
+        setItemViewDimensRadio()
+
+        if (calendarAdapter == null) {
+            calendarAdapter = PetCalendarAdapter()
+        } else {
+            calendarAdapter?.notifyDataSetChanged()
+        }
+    }
```

Comment:
```suggestion
        if (calendarAdapter == null) {
            calendarAdapter = PetCalendarAdapter()
        }
        calendarAdapter?.notifyDataSetChanged()
    }
```
else 문은 필요 없을 것 같아 제안드립니다 : )

Reply:
kotlin인데 자바 형식처럼 else문을 사용했네요. 감사합니다!

Reply:
```
if (calendarAdapter == null) {
            calendarAdapter = PetCalendarAdapter()
        }
calendarAdapter?.notifyDataSetChanged()
```

첫 라인에서 calendarAdapter 초기화하면 두번째 라인에서 calendarAdapter?.notifyDataSetChanged() 실행할때 ?는 빼도 동작되지 않나요? -> calendarAdapter.notifyDataSetChanged()

Reply:
![image](https://user-images.githubusercontent.com/33657541/104557272-5237ea00-5684-11eb-8ceb-01c48fbd44b9.png)

adapter 변수를 nullable, `var` 키워드로 선언해서 바로 위에서 초기화를 해주어도 밑에서 Null이 아님을 보장하지 않는 듯 합니다!